test/WriterTest.cpp

```diff
@@ -0,0 +1,60 @@
+#include <cstdio>
+#include <fstream>
+#include <iostream>
+
+#include "MaeBlock.hpp"
+#include "Reader.hpp"
+#include "Writer.hpp"
+#include "MaeConstants.hpp"
+
+#define BOOST_TEST_DYN_LINK
+#include <boost/test/unit_test.hpp>
+
+using namespace schrodinger::mae;
+using std::shared_ptr;
+
+BOOST_AUTO_TEST_SUITE(WriterSuite)
+
+BOOST_AUTO_TEST_CASE(Writer0)
+{
+    Reader r("test.mae");
+    auto w = new Writer("test_write.mae");
```

Comment:
use a unique_ptr instead of a raw ptr. (no make_unique until C++14, though)

Reply:
Switched to shared_ptr/reset, knowing there will only be one reference.